evaluate.go

```diff
@@ -224,9 +239,22 @@ func evaluateMatchExpression(expression *grammar.MatchExpression, datum interfac
 	}
 	val, err := ptr.Get(datum)
 	if err != nil {
-		if errors.Is(err, pointerstructure.ErrNotFound) && opts.withUnknown != nil {
-			err = nil
-			val = *opts.withUnknown
+		if errors.Is(err, pointerstructure.ErrNotFound) {
+			switch {
+			case opts.withUnknown != nil:
+				err = nil
+				val = *opts.withUnknown
+			case opts.withSkipMissingMapKey:
+				// For comparison results with missing keys: missing means not equal.
+				if evaluateSkipMissing(ptr, datum) {
+					switch expression.Operator {
+					case grammar.MatchEqual:
```

Comment:
I think we should probably handle the other operators too:

* MatchIn - false
* MatchNotIn - true
* MatchEmpty - true (sort of like how len(slice) == 0 works for a nil slice and an allocated slice that is currently empty)
* MatchNotEmpty - false
* MatchMatches - false
* MatchNotMatches - true

I would also recommend adding a func to grammar/ast.go (possibly called `NotPresentMatchDisposition`) to do the big switch statement over all operators and return the `bool, error` that you are returning in these cases. That way the big switch is colocated with other functions with similar needs and next to the const definitions so that when we add more operators in the future we don't accidentally miss adding them to this block of code.

Reply:
Any opinion on using an `Option` vs hardcoding the behavior? I think all HashiCorp products should have matching behavior, so I have a preference for hardcoding (or dropping this entirely and documenting the `"x" in m` approach if we can't get consensus).